activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -432,6 +432,7 @@ def in_order_of(column, values)
       references = column_references([column])
       self.references_values |= references unless references.empty?
 
+      values.map! { |value| type_caster.type_cast_for_database(column, value) }
```

Comment:
I'm not sure this should mutate the argument. It could be used for other things, and it results in odd `.dup` calls in your tests.

Reply:
@ghiculescu thank you very much for your review!
I agree that the `.dup` call in the tests is odd. I reverted the changes from `values.map!` to `values = values.map` and updated the CHANGELOG.